Grac superstructure

graciexie opened the pull request:
hey guys! i actually dont have that much changes to what banks had on saturday other than finishing up the statetrigs in superstructure and fixing the unit conversion stuff. im going to merge this branch into auto so i can have the right version of superstructure to use in autons ^-^

theoneandonlyr3n committed:
starting AutonChooser yippee

there are currently two displays showing auto chooser and auton chooser
need to troubleshoot and figure out how to get rid of autochooser

sohnshaik committed:
a bit more progress?

need to find more out on what specifically we are doing with this, and names are obv temp for now

theoneandonlyr3n committed:
created auton choosers for ele heights and coral station

theoneandonlyr3n committed:
created drop down for scoring locations

Bankst committed:
stuff

escapee-86 committed:
new auton convention names

theoneandonlyr3n committed:
wheel radius characterization

moved from flatbot

escapee-86 committed:
fixed variable names + more conventions

ayayrqn committed:
Merge branch 'autons' of https://github.com/WaltonRobotics/Reefscape into autons

escapee-86 committed:
fixed auton paths

escapee-86 committed:
capitalized acronyms

theoneandonlyr3n committed:
Merge branch 'autons' of https://github.com/WaltonRobotics/Reefscape into autons

theoneandonlyr3n committed:
restarting auton chooser

Commit:
updated trajsandlocs

partial commit for autonchooser schtuff

Commit:
oops i forgot LOLLERS

escapee-86 committed:
auton traj name fixes

sohnshaik committed:
fixing some errors w/ grac code

things in autonfactory are to be fixed later!!

sohnshaik committed:
Merge branch 'autons' of https://github.com/WaltonRobotics/Reefscape into autons

escapee-86 committed:
added trajs

Commit:
thsi stuffs completely unfinished but i figured id merge rn

Commit:
complete a merge the first time without messing up impossible edition

rhis SHOULD be right now. if i accidentally killed all ur progress oopsies

theoneandonlyr3n committed:
started working on firstScoring

Commit:
added new trgs to handle auton schtuff

Commit:
Merge branch 'autons' of https://github.com/WaltonRobotics/Reefscape into autons

Commit:
document thingy :/

sohnshaik committed:
almost there, minor fixes

theoretically finished by next meeting?! - also looking into DynamicSendableChooser and other things with it, will decide later rather than sooner
(yippee!)

sohnshaik committed:
Merge branch 'autons' of https://github.com/WaltonRobotics/Reefscape into autons

theoneandonlyr3n committed:
auton chooser work hurrah

also had to comment out the superstructure in robot cause errors

theoneandonlyr3n committed:
forgot to rename smth haha

Commit:
superstructure rewrite pt 1

the logic edition

Commit:
the spacing was bothering me

Commit:
MAN IT STILL DIDNT WORK

Commit:
idk this made more sense to me to include

Commit:
cleaned up coral

sohnshaik committed:
created HPStation on chooser :D

there is an ongoing problem with first scoring chooser, saying that the values have not been published to the network table

sohnshaik committed:
Merge branch 'autons' of https://github.com/WaltonRobotics/Reefscape into autons

theoneandonlyr3n committed:
created consumers cause maybe will utilize onChange()???

sohnshaik committed:
no more errors on elastic FINALLY

WE DID IT NO WAYYYYY(one line of code)

Commit:
im not finished but my comps abt to die so im stealing alexandras computer to cont working :3

theoneandonlyr3n committed:
continued working yayayay

Co-Authored-By: grape :D <90429497+graciexie@users.noreply.github.com>

sohnshaik committed:
final stretch started

started making the reefToHP and hpToReef

Commit:
trigs config'd

Commit:
banks did some things

Commit:
superstructure changes

Commit:
may have fixed conversions but i might be tweaking

Commit:
"i can do basic math, guys, trust!" I say, convincing nobody

but dw i fixed the conversions. im no longer tweaking. probably.

Commit:
Merge branch 'grac-superstructure' into autons

Bankst commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -96,7 +127,7 @@ private void configureBindings() {
         /*TODO: test to see if this actually works */
         // driver controls (not sure abt alignment inputs)
-        driver.rightTrigger().whileTrue(coral.score());
+        driver.rightTrigger().whileTrue(coral.setCoralMotorAction(kCoralSpeed));
```
what is this for?

Bankst commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -110,19 +141,21 @@ private void configureBindings() {
         manipulator.rightTrigger()
           .whileTrue(algae.setWheelAction(Algae.IntakeSpeed.PROCESSOR_SHOOT));
         manipulator.leftTrigger().and(manipulator.b())
-          .whileTrue(coral.intake());
+          .whileTrue(coral.setCoralMotorAction(kCoralSpeed));
```
what is this for?

Bankst commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -110,19 +141,21 @@ private void configureBindings() {
         // climber controls
-        manipulator.a().and(manipulator.povUp()).onTrue(elevator.setPosition(Elevator.EleHeights.CLIMB_UP));
-        manipulator.a().and(manipulator.povDown()).onTrue(elevator.setPosition(Elevator.EleHeights.CLIMB_DOWN));
+        // TODO: ask superstructure if in idle first
```
this still isn't done - either force the Superstructure into idle or check that it's at idle

Bankst commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -153,7 +197,7 @@ public void disabledExit() {}
   @Override
   public void autonomousInit() {
-    superstructure.autonPreload();
+    // superstructure.autonPreload();
```
this and the superstructure call below either need to be used or removed

Bankst commented on src/main/java/frc/robot/autons/WaltAutonFactory.java:
```diff
@@ -7,138 +7,95 @@
+                    new Elastic.Notification(
+                        Elastic.Notification.NotificationLevel.ERROR, 
+                        "Invalid reef to HP AutonCycle", 
+                        "UR AUTON SUX AND HAS AN INVALID REEF-TO-HP YOU DUMB funnel cake"
```
Somewhat rude

Bankst commented on src/main/java/frc/robot/autons/WaltAutonFactory.java:
```diff
@@ -7,138 +7,95 @@
-        return m_routine;
-    }
-}
+    // public Command generateAuton(Swerve drivetrain, Superstructure superstructure, ArrayList<AutonCycle> cycles) { // im assuming that autoncycle will include 1st iteration
+    //     // iterate AutonCycles and validate all HP->Score and Score->HP pairs
+    //     for (AutonCycle autonCycle : cycles) {
+    //         if(!autonCycle.isLegit()) {
+    //             return Commands.print("that one scene in alice in borderland where karube just sits there contemplatively before he blows up.");
+    //         }
+    //     }
+
+    //     AutoTrajectory firstScoreTraj = m_routine.trajectory(Trajectories.ReefToHPTrajs.get(cycles.get(0).reefHPPair));
+    //     AutoTrajectory firstLoadTraj = m_routine.trajectory(Trajectories.HPToReefTrajs.get(cycles.get(0).hpReefPair));
+
+    //     m_routine.active().onTrue(
+    //         Commands.sequence(
+    //             firstScoreTraj.resetOdometry(),
+    //             firstScoreTraj.cmd()
+    //         )
+    //     );
+
+    //     firstScoreTraj.atTime("eleUp").onTrue()
+    // }
+}
```
If this is unused remove it

Bankst commented on src/main/java/frc/robot/subsystems/Coral.java:
```diff
@@ -69,28 +65,15 @@ public void setFingerCoast(boolean coast) {
     public Command setCoralMotorAction(double destinationVelocity) {
-        m_coralIsRunning = true;
         return Commands.runEnd(
             () -> m_coralMotor.set(destinationVelocity),
             () -> m_coralMotor.set(0)
```
this will not work - the coral motor will never run.

Bankst commented on src/main/java/frc/robot/subsystems/Superstructure.java:
```diff
@@ -10,137 +11,197 @@
             () -> {
-                if(!m_driverRumbled) {
+                if(!driverRumbled) {
                     m_driverRumbler.accept(intensity);
-                    m_driverRumbled = true;
+                    driverRumbled = true;
```
What is the purpose of this flag?

Commit:
waltautonfactory should be doneee

Commit:
waltautonfactory should be doneee

Commit:
waltautonfactory should be doneee

Commit:
ill fix robot.java in todays meeting prolly - its REALLY brokey.

but for now i cleaned up everythign else.

Commit:
made warnings nicer.

aka killed its personality IT WAS SO GOOD BEFORE UGH

Commit:
useless update BUT when i said autonfactory was done, i meant it was done cept stuff i need from autonchooser.

Commit:
meow !

sohnshaik committed:
made some more progress, will most likely have to change tho

sohnshaik committed:
Merge branch 'autons' of https://github.com/WaltonRobotics/Reefscape into autons

Commit:
just random but added commands.print for state changes

theoneandonlyr3n committed:
finished with first cycle of auton for autonchooser

Commit:
privatized vars

Commit:
privatized vars

sohnshaik committed:
started working on cycle funsies

sohnshaik committed:
Merge branch 'autons' of https://github.com/WaltonRobotics/Reefscape into autons

Commit:
cleaned up logging state process.

i should add the like. add the graph thing. but i should also finish algae schtuff first so ill do that after

Bankst commented on src/main/java/frc/robot/autons/WaltAutonFactory.java:
```diff
@@ -7,138 +7,153 @@
+        );
+
+        firstScoreTraj.atTime("eleUp")
+            .onTrue(Commands.runOnce(() -> superstructure.requestIsPreload(true))
+            .andThen(superstructure.autonRequestEleToScore(firstHeight))
+            .andThen(Commands.runOnce(() -> superstructure.requestIsPreload(false))));
```
don't chain multiple andThens, use a Sequence

Bankst commented on src/main/java/frc/robot/autons/WaltAutonFactory.java:
```diff
@@ -7,138 +7,153 @@
-        for (int i = 0; i < m_trajList.size(); i++) {
-            m_cmdSched.addCommands(
-                m_trajList.get(i).resetOdometry(), // im actually a tad unsure if ur supposed to reset odometry before every path or j the first one. will find out at some pt  
+        AutoTrajectory firstScoreTraj = m_routine.trajectory("start to reef");
+        AutoTrajectory firstLoadTraj = m_routine.trajectory("reef to hp");
```
even if not yet functional, these should pull from the correct chooser/list/etc.

graciexie replied:
that doesnt exist yet, but sohan said he'll get it done by likeee tonight probably hopefully

Bankst commented on src/main/java/frc/robot/autons/WaltAutonFactory.java:
```diff
@@ -7,138 +7,153 @@
-                i++;
-                m_cmdSched.addCommands(
-                    m_trajList.get(i).resetOdometry(),
+            hpToReefTraj.atTime("eleUp").onTrue(superstructure.autonRequestEleToScore(cycles.get(cycleIdx).height));
```
```suggestion
            hpToReefTraj.atTime("eleUp").onTrue(superstructure.autonRequestEleToScore(cycle.height));
```

Bankst commented on src/main/java/frc/robot/autons/WaltAutonFactory.java:
```diff
@@ -7,138 +7,153 @@
-        return m_cmdSched;
-    }
+            reefToHpTraj.atTime("intake")
+                .onTrue(superstructure.autonRequestToIntake()
+                .alongWith(Commands.print("RunTheIntakePleaseeeeeeeeeeeeee")));
```
prefer sequence to andthen

graciexie replied:
dyou want me to change the alongWith -> parallel?

Bankst replied:
yes

Bankst commented on src/main/java/frc/robot/autons/WaltAutonFactory.java:
```diff
@@ -7,138 +7,153 @@
-        return m_cmdSched;
-    }
+            reefToHpTraj.atTime("intake")
+                .onTrue(superstructure.autonRequestToIntake()
+                .alongWith(Commands.print("RunTheIntakePleaseeeeeeeeeeeeee")));
```
log using a double incrementer instead of a string, like previous years

Bankst commented on src/main/java/frc/robot/subsystems/Superstructure.java:
```diff
@@ -1,146 +1,235 @@
-            (stateTrg_scoring.and(() -> !m_coral.bs_coralMotorRunning.getAsBoolean())).onTrue(Commands.runOnce(() -> m_coralState = CoralState.IDLE));
-        }
+    private void configureAutonTrgs() {
+        // these are treated kinda differently in auton than in teleop so i differentiated them again
```
if the state actions differ between modes, they need to be documented, and &'d with RobotModeTriggers.autonomous (the others being &'d with teleop), unless we are positive they can never fire at the same time

graciexie replied:
the state actions don't differ, but all of the separated out triggers are button-binded in teleop. Thus the separation. And then the auton triggers would only ever be called in autonomous, but I added in RobotModeTrigger.autonomous to them just in case.

Bankst commented on src/main/java/frc/robot/subsystems/Superstructure.java:
```diff
@@ -1,146 +1,235 @@
+        return Commands.run(
+           () -> m_driverRumbler.accept(0)
+        ).withTimeout(secs);
     }
 
-    public void resetAfterAuton() {
-        m_preload = false;
-        m_autonScoreReq = false;
+    private Command manipRumble(double intensity, double secs) {
+        return Commands.run(
+            () -> m_manipRumbler.accept(0)
+        ).withTimeout(secs);
```
the intensity is not being passed to the rumbler

graciexie replied:
whoops

theoneandonlyr3n committed:
got cycles to work kind of?

the options displayed for the cycles are not of the cycles before it, but of the original HP chooser so will hopefully fix that tonight

sohnshaik committed:
the final push there

2 errors remaining, publisher problem and HP station problem, should get them fixed

Commit:
cleaned up superstructure.

this commit will get moved

Commit:
also fixed nte names here once and for all. HELP ive lit changed this same thing on like ten thousand branches lmaoo

also btw as a side note im getting rid of the coral and ele branches cuz all changes should j be done in superstructure.

theoneandonlyr3n committed:
i think i fixed publisher updae error!

Commit:
commented out robot FOR NOW

the errors were bothering me :'(

Commit:
Merge branch 'autons' into grac-superstructure

Commit:
fixed multipe .andThen problem

Commit:
fixed banks requested change

Commit:
lol im stupid. now the superstructures properly logified.

Commit:
implemented banks changes on the pull req

Bankst commented on src/main/java/frc/robot/Main.java:
```diff
@@ -10,6 +10,6 @@ public final class Main {
 
   public static void main(String... args) {
-    RobotBase.startRobot(Robot::new);
+    // RobotBase.startRobot(Robot::new);
```
Absolutely do not.

Bankst commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -1,197 +1,249 @@
-        driver.rightTrigger().whileTrue(algae.setWheelAction(Algae.IntakeSpeed.PROCESSOR_SHOOT));
+/* NOTE: a lot of the methods we had in the button bindings now dont exist. ill go back and fix them as soon as im done fixing all the other issues in the classes themselves
+ * but
+ * the errors were bothering me so i killed robot.java for now.
```
Do not do this.

Bankst commented on src/main/java/frc/robot/autons/WaltAutonFactory.java:
```diff
@@ -7,138 +7,156 @@
+                    new Elastic.Notification(
+                        Elastic.Notification.NotificationLevel.ERROR, 
+                        "Invalid reef to HP AutonCycle", 
+                        "INVALID REEF-TO-HP: if this auton was a cat, you just killed it."
```
Why are we killing cats?

Commit:
uh fixed alongwiths and uncommented out robot.java

robot.java still has errors tho

Commit:
got rid of the problems in robot.java. ill populate buttonbindings again later.

i first have to fix climb. and ill prob implement autons too.

Commit:
added climbing and elevator override mode to superstructure

Commit:
my bad gang
